portal-impl/src/com/liferay/portal/model/impl/UserImpl.java

```diff
@@ -845,7 +845,23 @@ public boolean isPasswordReset() {
 			return false;
 		}
 
-		return getPasswordReset();
+		boolean updatePasswordRequired = false;
+
+		try {
+			Company company = CompanyLocalServiceUtil.getCompany(
+				getCompanyId());
+
+			updatePasswordRequired = company.isUpdatePasswordRequired();
```

Comment:
@cdbm @ChrisKian Why do we need to check this company setting?
I think it is for a different purpose...
```
    #
    # Set this to true to require users to input their password before email
    # address or screen name updates.
    #
    # Env: LIFERAY_COMPANY_PERIOD_SECURITY_PERIOD_UPDATE_PERIOD_PASSWORD_PERIOD_REQUIRED
    #
    company.security.update.password.required=true
```

Reply:
You're right, I meant to use the password policy to determine if a change is required.  I'll update the PR now